site/source/tutorials/007_theming/index.md

```diff
@@ -0,0 +1,149 @@
+
+{% instruction 'Theme the Worker widget' %}
+
+In order to theme the `Worker` widget, we need to create `worker.m.css` within our theme directory and use it within `theme.ts`. The naming here is important as the object key of the exported theme must match the naming of the widget's CSS file.
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/theme.ts' lines:3,9,11,16 %}
+
+Let's start by creating a red `Worker` and observe our theme being applied correctly.
+
+```css
+/* worker.m.css */
+.root {
+	background: red;
+}
+```
+
+To apply a theme to a widget, simply pass the `theme` as a property to widgets have the `ThemeableMixin` applied. To ensure that the entire application applies the `theme` it needs to be passed to all the themeable widgets in our application. This can become problematic when an application using a mixture of themeable and non-themeable widgets or uses a widget from a third party, meaning that there is no guarantee that the `theme` will be propagated as required.
+
+{% aside 'What is a registry?' %}
+A registry provides a mechanism to inject external payloads into widgets throughout the application tree. To learn more, take a look at the [container tutorial](../1010_containers_and_injecting_state/) and [registry tutorial](../1020_registries/).
+{% endaside %}
+
+However an application can automatically inject a `theme` from the `registry` to every themeable widget in an application tree. First, create a `themeInjector` using a the `registerThemeInjector` by passing a `registry` instance and `theme`. This will returns a handle to the `themeInjector` that can be used to change the theme using `themeInjector.set()`, which will invalidate all themeable widgets in the application tree and re-render using the new theme!
+
+{% instruction 'Update our `main.ts` file to import our theme and create a `themeInjector`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' %}
+
+Open the application in your web browser and see that the `Worker` backgrounds are now `red`.
+
+{% instruction 'Use variables and complete the Worker theme' %}
+
+The [Dojo 2 build system](../006_deploying_to_production/) supports new CSS features such as `css-custom-properties` by using PostCSS to process our `.m.css` files. We can use these new CSS features to add variables to `worker.m.css` and complete its theme.
+Let's create `themes/dojo/variables.css` (notice that this file does not have a `.m.css` extension as it is not a `css-module` file).
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/variables.css' lang:css %}
+
+We can now use these variables in `worker.m.css` to create our fully themed `Worker`.
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/worker.m.css' lang:css %}
```

Comment:
Should have a couple of statements here - one summarizing what we did in this step and one foreshadowing what is coming up.